test/integration/node-specific/resource_tracking_script_builder.ts

```diff
@@ -0,0 +1,132 @@
+import { fork } from 'node:child_process';
+import { on, once } from 'node:events';
+import { readFile, unlink, writeFile } from 'node:fs/promises';
+import * as path from 'node:path';
+
+import { expect } from 'chai';
+import { parseSnapshot } from 'v8-heapsnapshot';
+
+import { MongoClient } from '../../mongodb';
+import { TestConfiguration } from '../../tools/runner/config';
+
+export type ResourceTestFunction = (options: {
+  MongoClient: typeof MongoClient;
+  uri: string;
+  iteration: number;
+}) => Promise<void>;
+
+export const testScriptFactory = (
+  name: string,
+  uri: string,
+  iterations: number,
+  func: ResourceTestFunction
+) => `'use strict';
```

Comment:
I've been thinking about how to make this script itself easier to maintain, and I wonder if we could factor it out into a stand-alone `.js` file, where the things currently in `${}` scopes are marked with `'CLIENT_PATH'`, `'FUNCTION'` and `'URI'` respectively, then you could read this code in from the corresponding file here, and use a simple `.replace` to insert the `JSON.stringify` and `.toString()` values into the file.

Reply:
Broke it out to a file with replace calls, let me know how it looks maintainability wise

Reply:
I think that works, thanks!